### DISCUSSION TITLE
Is there a "Concurrency" feature in Serenity?

### DISCUSSION THREAD

**Discussion by axelcasauran** [serenity-is/Serenity#6295] [General]
Hi,

I was wondering if there is a concurrency feature in the serenity where if the record is open in user #1 and was modified at the same time in user #2, how can we handle it? or is serenity framework already handled it?

**Comment by VictorTomaili** [comment]
Serenity is not handling this type of situation. You need to cache some lock status and retrieve this status together with the entity itself. if the entity is locked, you are not gonna allow editing.

**Reply by axelcasauran** [reply]
Thank you for the fast reply. I will look for a solution on this. Thanks.

**Comment by minhhungit** [comment]
try to use [signalR](https://docs.microsoft.com/en-us/aspnet/signalr/overview/getting-started/introduction-to-signalr)

there are some sample code in wiki of this repository you can review

![image](https://user-images.githubusercontent.com/2279508/157890871-edda4a04-b88e-4291-8422-67d83708560f.png)
